Rules/UseCompatibleCmdlets.cs

```diff
@@ -12,343 +12,408 @@
+                }
+
+                dynamic deserialized = JObject.Parse(File.ReadAllText(file));
+                psCmdletMap[fileNameWithoutExtension] = GetCmdletsFromData(deserialized);
+            }
+        }
 
-                // ignore (warn) invalid entries
-                if (GetVersionInfoFromPlatformString(compat, out psedition, out psversion, out os))
+        /// <summary>
+        /// Get a hashset of cmdlet names from a deserialized json file
+        /// </summary>
+        private HashSet<string> GetCmdletsFromData(dynamic deserializedObject)
+        {
+            HashSet<string> cmdlets = new HashSet<string>(StringComparer.OrdinalIgnoreCase);
+            dynamic modules = deserializedObject.Modules;
+            foreach (dynamic module in modules)
+            {
+                if (module.ExportedCommands == null)
+                {
+                    continue;
+                }
+
+                foreach (dynamic cmdlet in module.ExportedCommands)
                 {
-                    platformSpecMap.Add(compat, new { PSEdition = psedition, PSVersion = psversion, OS = os });
-                    curCmdletCompatibilityMap.Add(compat, true);
+                    string name = cmdlet.Name.ToString();
+                    cmdlets.Add(name);
                 }
             }
+            return cmdlets;
+        }
+
+        /// <summary>
+        /// Set up cmdlet list for Linux/OSX known issues.
+        /// </summary>
+        private List<string> setUpKnownIssuesList(string path)
+        {
+            var knownIssuesFile = Directory.GetFiles(path, knownIssuesFileName);
+            dynamic deserialized = JArray.Parse(File.ReadAllText(knownIssuesFile[0]));
+            List<string> issues = new List<string>();
 
-            ProcessDirectory(
-                settingsPath,
-                extentedCompatibilityList);
-            if (psCmdletMap.Keys.Count != extentedCompatibilityList.Count())
+            foreach (dynamic cmdlet in deserialized)
             {
-                return;
+                issues.Add(cmdlet.ToString());
             }
 
-            // reached this point, so no error
-            hasInitializationError = false;
+            return issues;
+        }
+
+        /// <summary>
+        /// Set up cmdlet map from the latest desktop version of PowerShell.
+        /// </summary>
+        private HashSet<string> SetUpReferenceCmdletMap(string path, string fileName)
+        {
+            string[] cmdletFile = Directory.GetFiles(path, fileName);
+            dynamic deserialized = JObject.Parse(File.ReadAllText(cmdletFile[0]));
+            return GetCmdletsFromData(deserialized);
+        }
+
+        /// <summary>
+        /// Check if cmdlet is on the known Issues List for Linux/OSX.
+        /// </summary>
+        private bool checkKnownIssuesList(string commandName)
+        {
+            if (knownIssuesList.Contains(commandName, StringComparer.OrdinalIgnoreCase))
```

Comment:
Should be able to just return `knownIssuesList.Contains(commandName, StringComparer.OrdinalIgnoreCase)` here without any `if` statement